q.js

```diff
@@ -1502,6 +1502,10 @@ Promise.prototype.keys = function () {
 Q.all = all;
 function all(promises) {
     return when(promises, function (promises) {
+        if (typeof arguments[0].length === "undefined") {
```

Comment:
With Array.isArray, if all() gets passed the arguments object or a function that would eventually return an array of promises, it would throw an error.